File: IsFibo/IsFibo.java
import java.util.Scanner;


public class IsFibo {
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int nNumbers = sc.nextInt();
		long[] fibSequence = {1,1,1};
		while(nNumbers-- > 0){
			long number = sc.nextLong();
			if(number < fibSequence[2]){
				fibSequence[0] = 1;
				fibSequence[1] = 1;
				fibSequence[2] = 1;
			}
			while(fibSequence[2] < number){
				fibSequence[2] = fibSequence[0] + fibSequence[1];
				fibSequence[0] = fibSequence[1];
				fibSequence[1] = fibSequence[2];
			}
			if(fibSequence[2] == number) System.out.println("IsFibo");
			else System.out.println("IsNotFibo");
		}
	}

}

File: IsFibo/isFibo.py
'''
Created on 30/07/2014

@author: miguel
'''

if __name__ == '__main__':
    
    nNumbers = int(input())
    fibs = [1,1,1]
    for i in range(nNumbers):
        number = int(input())
        if(fibs[2] > number):
            fibs = [1,1,1]
        while(fibs[2] < number):
            fibs[2] = fibs[0] + fibs[1]
            fibs = [fibs[1], fibs[2], fibs[2]]
        if(fibs[2] == number):
            print('IsFibo')
        else: print('IsNotFibo')